pkg/controller/controller.go

```diff
@@ -284,6 +284,15 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
 				return nil, status.Errorf(codes.InvalidArgument, "invalid RPO value")
 			}
 
+			// Validating RPO to be non Zero when replication mode is ASYNC
+			if repMode == "ASYNC" && rpo == "Zero" {
+				return nil, status.Errorf(codes.InvalidArgument, "replication mode ASYNC requires RPO value to be non Zero")
+			}
+
+			// Validating RPO to be Zero whe replication mode is SYNC
```

Comment:
Hi @rajendraindukuri 
Can we make the RPO input as **optional** for SYNC mode? We can validate it for 'Zero' only if it is specified.

@donatwork @lukeatdell @karthikk92 @alankar-verma : Any thoughts on this?
To give you some background on this: PowerStore API only takes in RPO value for replication rule creation (0 is considered as Sync and other values as Async). PowerStore UI disables RPO input once user selects Sync mode.
In our StorageClass, we have introduced a new parameter 'mode' for sync, async or metro. So, user has to specify the mode anyway. He doesn't need to provide RPO for Sync mode.

Reply:
Make sense to me. Just need to validate with csi-powerstore.

Reply:
@santhoshatdell Updated the RPO required validatoin to throw error when RPO is not passed with Replicaiton mode ASYNC and also set RPO to zero when Replication mode is SYNC. Updated unit tests for the changes done. Thanks.